### DISCUSSION TITLE
New developer focused workflow article about how the login and registration works in WordPress.

### DISCUSSION THREAD

**Discussion by Rajinsharwar** [WordPress/developer-blog-content#159] [Topic Ideas]
When developing user authentication systems for WordPress, it's really important to have a deep understanding of the login and registration processes. I am proposing a new article that will shed some light on the intricacies of these processes, and the underlying functions that ensure secure and seamless user experiences in WordPress sites. 

I have also created a draft documentation for the purpose. Requesting your opinion and review on this:
https://docs.google.com/document/d/17N4uVctGa1pk8B2AcqY7KGGlY6yVV9azn85TsGJGALA/edit?usp=drivesdk

**Comment by bph** [comment]
Hi @Rajinsharwar - Thank you so much for your topic idea and providing a draft already!  This seems to be a great tutorial covering in details the following 

- How does logging in work in WordPress?
- Modifying the login processes.
- How does registration work in WordPress?
- Modifying the registration processes.
- FAQs regarding login and registration.

**Comment by justintadlock** [comment]
Is the draft meant to be documentation that you will use as the basis for the article or is it the proposal for the article? It wasn't 100% clear. Regardless, it is an interesting topic.

My response below assumes that you're proposing this as a draft for an article:

The seems like a good informational article, but I find myself wanting to know more. There are entire sections on modifying the login and registration process, but we never actually dive into those subjects deeply or see any code examples.

I understand that this isn't really a how-to guide, but that's where I'd want it to lead me as a reader---either in a Part 2 post or within this post.

Even if it's not a how-to, I still think you could use code examples to show how specific things work. For example, what are the login action/filter hooks? How can I add actions/filters to modify the process? Or, how can I leverage the functions you mentioned?

I think it's a good start and would love to see the topic explored on the Dev Blog.

**Reply by Rajinsharwar** [reply]
Hi @justintadlock, the draft is just for proposing the idea of the how-to post topic. Yeah, if the topic is accepted, I am happy to extend it with more explanations and code examples. 🙂

**Comment by bph** [comment]
This discussion is locked. Please continue conversation on the issue 

- #165

**Comment by bph** [comment]
Published:[How do Log in and Registration work for developers?](https://developer.wordpress.org/news/2024/02/12/how-do-log-in-and-registration-work-for-developers/)
